File: 822-unique-morse-code-words/unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] str={".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        HashSet<String> s=new HashSet<>();
        for(String st:words)
        {
            StringBuffer sb=new StringBuffer();
              for(int i=0;i<st.length();i++)
              {
                sb.append(str[st.charAt(i)-'a']);
              }
                s.add(sb.toString());
              
        }
        return s.size();
    }
}

File: 822-unique-morse-code-words/unique-morse-code-words.php
class Solution {

    /**
     * @param String[] $words
     * @return Integer
     */
    function uniqueMorseRepresentations($words) {
        $store=array(".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--..");
        $has=[];
        foreach($words as $key)
        {
            $res="";
            for($i=0;$i<strlen($key);$i++)
            {
               $index=ord($key[$i])-97;
               $res.=$store[$index];
            }
            $has[$res]=true;
        }
        return count($has);
    }
}